src/helpers/files.js

```diff
@@ -92,7 +77,7 @@ const registerFileActionFallback = () => {
 				const file = window.FileList.findFile(filename)
 				Promise.all([
 					import('vue'),
-					import(/* webpackChunkName: "files-modal" */'./../components/PublicFilesEditor'),
+					import(/* webpackChunkName: "files-modal" */'@nextcloud/text-editor/components/PublicFilesEditor'),
```

Comment:
I guess we will change this approach.

Reply:
But how... this is really one of the parts i have no idea of.

I think we will still want to have async imports. But we don't want to import from a nested path inside the package but from the package itself i think.

Reply:
This is hard do to in actual versions of webpack 😞 
But we can try to create local files to make references to direct components on `'@nextcloud/text-editor'` and lazy load those files.


```js 
// src/components/editor.js

const Editor = () => import(/* webpackChunkName: "editor" */'@nextcloud/text-editor') 

export const PublicFilesEditor = () => Editor().then(package => package.PublicFilesEditor)
```

That is not the best solution, and we need to test how webpack will handle it, but I guess we can try.

Reply:
We also need to take into account the bundle size of the various webpack chunks. I tried making the `EditorWrapper` the default export of the package - but that made all chunks that used the `{ state }` import from the same package grow by more than 1 MB. So right now i think i'd go with just using file references from inside the alias and try and separate the code into the src and package dirs in a meaningful way.